// @react-pdf-viewer/default-layout/lib/cjs/default-layout.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('@react-pdf-viewer/attachment/lib/styles/index.css');
require('@react-pdf-viewer/bookmark/lib/styles/index.css');
require('@react-pdf-viewer/core/lib/styles/index.css');
require('@react-pdf-viewer/thumbnail/lib/styles/index.css');
require('@react-pdf-viewer/toolbar/lib/styles/index.css');
var React = require('react');
var core = require('@react-pdf-viewer/core');
var attachment = require('@react-pdf-viewer/attachment');
var bookmark = require('@react-pdf-viewer/bookmark');
var thumbnail = require('@react-pdf-viewer/thumbnail');
var toolbar = require('@react-pdf-viewer/toolbar');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

var BookmarkIcon = function () {
    return (React__default['default'].createElement(core.Icon, { size: 16 },
        React__default['default'].createElement("path", { d: "M11.5,1.5h11c0.552,0,1,0.448,1,1v20c0,0.552-0.448,1-1,1h-21c-0.552,0-1-0.448-1-1v-20c0-0.552,0.448-1,1-1h3\n                M11.5,10.5c0,0.55-0.3,0.661-0.659,0.248L8,7.5l-2.844,3.246c-0.363,0.414-0.659,0.3-0.659-0.247v-9c0-0.552,0.448-1,1-1h5\n                c0.552,0,1,0.448,1,1L11.5,10.5z\n                M14.5,6.499h6\n                M14.5,10.499h6\n                M3.5,14.499h17\n                M3.5,18.499h16.497" })));
};

/*! *****************************************************************************
Copyright (c) Microsoft Corporation.

Permission to use, copy, modify, and/or distribute this software for any
purpose with or without fee is hereby granted.

THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
PERFORMANCE OF THIS SOFTWARE.
***************************************************************************** */

var __assign = function() {
    __assign = Object.assign || function __assign(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};

var FileIcon = function () {
    return (React__default['default'].createElement(core.Icon, { size: 16 },
        React__default['default'].createElement("path", { d: "M7.618,15.345l8.666-8.666c0.78-0.812,2.071-0.838,2.883-0.058s0.838,2.071,0.058,2.883\n                c-0.019,0.02-0.038,0.039-0.058,0.058L7.461,21.305c-1.593,1.593-4.175,1.592-5.767,0s-1.592-4.175,0-5.767c0,0,0,0,0,0\n                L13.928,3.305c2.189-2.19,5.739-2.19,7.929-0.001s2.19,5.739,0,7.929l0,0L13.192,19.9" })));
};

var ThumbnailIcon = function () {
    return (React__default['default'].createElement(core.Icon, { size: 16 },
        React__default['default'].createElement("path", { d: "M10.5,9.5c0,0.552-0.448,1-1,1h-8c-0.552,0-1-0.448-1-1v-8c0-0.552,0.448-1,1-1h8c0.552,0,1,0.448,1,1V9.5z\n                M23.5,9.5c0,0.552-0.448,1-1,1h-8c-0.552,0-1-0.448-1-1v-8c0-0.552,0.448-1,1-1h8c0.552,0,1,0.448,1,1V9.5z\n                M10.5,22.5 c0,0.552-0.448,1-1,1h-8c-0.552,0-1-0.448-1-1v-8c0-0.552,0.448-1,1-1h8c0.552,0,1,0.448,1,1V22.5z\n                M23.5,22.5c0,0.552-0.448,1-1,1 h-8c-0.552,0-1-0.448-1-1v-8c0-0.552,0.448-1,1-1h8c0.552,0,1,0.448,1,1V22.5z" })));
};

var TOOLTIP_OFFSET = { left: 8, top: 0 };
var Sidebar = function (_a) {
    var tabContents = _a.tabContents;
    var l10n = React.useContext(core.LocalizationContext);
    var _b = React.useState(false), opened = _b[0], setOpened = _b[1];
    var _c = React.useState(0), tab = _c[0], setTab = _c[1];
    var switchToTab = function (index) {
        if (tab === index) {
            setOpened(function (isOpened) { return !isOpened; });
        }
        else {
            setOpened(true);
            setTab(index);
        }
    };
    return (React__default['default'].createElement("div", { className: "rpv-default-layout-sidebar " + (opened ? 'rpv-default-layout-sidebar-opened' : '') },
        React__default['default'].createElement("div", { className: 'rpv-default-layout-sidebar-tabs' },
            React__default['default'].createElement("div", { className: 'rpv-default-layout-sidebar-headers' },
                React__default['default'].createElement("div", { className: 'rpv-default-layout-sidebar-header' },
                    React__default['default'].createElement(core.Tooltip, { position: core.Position.RightCenter, target: (React__default['default'].createElement(core.Button, { onClick: function () { return switchToTab(0); }, isSelected: tab === 0 },
                            React__default['default'].createElement(ThumbnailIcon, null))), content: function () {
                            return l10n && l10n.defaultLayout ? l10n.defaultLayout.thumbnail : 'Thumbnail';
                        }, offset: TOOLTIP_OFFSET })),
                React__default['default'].createElement("div", { className: 'rpv-default-layout-sidebar-header' },
                    React__default['default'].createElement(core.Tooltip, { position: core.Position.RightCenter, target: (React__default['default'].createElement(core.Button, { onClick: function () { return switchToTab(1); }, isSelected: tab === 1 },
                            React__default['default'].createElement(BookmarkIcon, null))), content: function () {
                            return l10n && l10n.defaultLayout ? l10n.defaultLayout.bookmark : 'Bookmark';
                        }, offset: TOOLTIP_OFFSET })),
                React__default['default'].createElement("div", { className: 'rpv-default-layout-sidebar-header' },
                    React__default['default'].createElement(core.Tooltip, { position: core.Position.RightCenter, target: (React__default['default'].createElement(core.Button, { onClick: function () { return switchToTab(2); }, isSelected: tab === 2 },
                            React__default['default'].createElement(FileIcon, null))), content: function () {
                            return l10n && l10n.defaultLayout ? l10n.defaultLayout.attachment : 'Attachment';
                        }, offset: TOOLTIP_OFFSET }))),
            React__default['default'].createElement("div", { className: "rpv-default-layout-sidebar-content " + (opened ? 'rpv-default-layout-sidebar-content-opened' : '') }, tabContents && tabContents[tab] ? tabContents[tab]() : React__default['default'].createElement(React__default['default'].Fragment, null)))));
};

var defaultLayoutPlugin = function (props) {
    var attachmentPluginInstance = attachment.attachmentPlugin();
    var bookmarkPluginInstance = bookmark.bookmarkPlugin();
    var thumbnailPluginInstance = thumbnail.thumbnailPlugin();
    var toolbarPluginInstance = toolbar.toolbarPlugin(props ? props.toolbarPlugin : {});
    var Attachments = attachmentPluginInstance.Attachments;
    var Bookmarks = bookmarkPluginInstance.Bookmarks;
    var Thumbnails = thumbnailPluginInstance.Thumbnails;
    var Toolbar = toolbarPluginInstance.Toolbar;
    var plugins = [
        attachmentPluginInstance,
        bookmarkPluginInstance,
        thumbnailPluginInstance,
        toolbarPluginInstance,
    ];
    return {
        install: function (pluginFunctions) {
            plugins.forEach(function (plugin) {
                if (plugin.install) {
                    plugin.install(pluginFunctions);
                }
            });
        },
        renderViewer: function (renderProps) {
            var slot = renderProps.slot;
            plugins.forEach(function (plugin) {
                if (plugin.renderViewer) {
                    slot = plugin.renderViewer(__assign(__assign({}, renderProps), { slot: slot }));
                }
            });
            var mergeSubSlot = slot.subSlot && slot.subSlot.attrs
                ? {
                    ref: slot.subSlot.attrs.ref,
                    style: slot.subSlot.attrs.style
                }
                : {};
            slot.children = (React__default['default'].createElement("div", { className: 'rpv-default-layout-container' },
                React__default['default'].createElement("div", { className: 'rpv-default-layout-toolbar' }, props && props.renderToolbar
                    ? props.renderToolbar(Toolbar)
                    : React__default['default'].createElement(Toolbar, null)),
                React__default['default'].createElement("div", { className: 'rpv-default-layout-main' },
                    React__default['default'].createElement(Sidebar, { tabContents: [
                            function () { return React__default['default'].createElement(Thumbnails, null); },
                            function () { return React__default['default'].createElement(Bookmarks, null); },
                            function () { return React__default['default'].createElement(Attachments, null); },
                        ] }),
                    React__default['default'].createElement("div", __assign({ className: 'rpv-default-layout-body' }, mergeSubSlot), slot.subSlot.children)),
                slot.children));
            slot.subSlot.attrs = {};
            slot.subSlot.children = React__default['default'].createElement(React__default['default'].Fragment, null);
            return slot;
        },
        uninstall: function (pluginFunctions) {
            plugins.forEach(function (plugin) {
                if (plugin.uninstall) {
                    plugin.uninstall(pluginFunctions);
                }
            });
        },
        onDocumentLoad: function (props) {
            plugins.forEach(function (plugin) {
                if (plugin.onDocumentLoad) {
                    plugin.onDocumentLoad(props);
                }
            });
        },
        onTextLayerRender: function (props) {
            plugins.forEach(function (plugin) {
                if (plugin.onTextLayerRender) {
                    plugin.onTextLayerRender(props);
                }
            });
        },
        onViewerStateChange: function (viewerState) {
            var newState = viewerState;
            plugins.forEach(function (plugin) {
                if (plugin.onViewerStateChange) {
                    newState = plugin.onViewerStateChange(newState);
                }
            });
            return newState;
        },
    };
};

exports.BookmarkIcon = BookmarkIcon;
exports.FileIcon = FileIcon;
exports.ThumbnailIcon = ThumbnailIcon;
exports.defaultLayoutPlugin = defaultLayoutPlugin;
